core/src/main/scala/com/salesforce/op/utils/stages/NameDetectUtils.scala

```diff
@@ -0,0 +1,337 @@
+        if (genderResult == GenderNA) 1 else 0
+      )
+    } toMap
+  }
+
+  private[op] def computeResults(
+    input: String,
+    nameDict: Broadcast[NameDictionary],
+    genderDict: Broadcast[GenderDictionary],
+    hll: HyperLogLogMonoid
+  ): NameDetectStats = input match {
+    case "" if $(ignoreNulls) => NameDetectStats.empty
+    case _ =>
+      val tokens = preProcess(input)
+      NameDetectStats(
+        computeGuardCheckQuantities(input, tokens, hll),
+        AveragedValue(1L, dictCheck(tokens, nameDict.value)),
+        computeGenderResultsByStrategy(input, tokens, genderDict.value)
+      )
+  }
+
+  type NameDetectMapFun[T <: Text] = T#Value => NameDetectStats
+  private[op] def makeMapFunction[T <: Text](spark: SparkSession): NameDetectMapFun[T] = {
+    val broadcastNameDict: Broadcast[NameDictionary] = spark.sparkContext.broadcast(DefaultNameDictionary)
+    val broadcastGenderDict: Broadcast[GenderDictionary] = spark.sparkContext.broadcast(DefaultGenderDictionary)
+    val hllMonoid = new HyperLogLogMonoid(NameDetectUtils.HLLBits)
+
+    computeResults(_, broadcastNameDict, broadcastGenderDict, hllMonoid)
+  }
+
+  private[op] def computeTreatAsName(results: NameDetectStats): Boolean = {
+    val hllMonoid = new HyperLogLogMonoid(NameDetectUtils.HLLBits)
+    val guardChecksPassed = performGuardChecks(results.guardCheckQuantities, hllMonoid)
+    val predictedNameProb = results.dictCheckResult.value
+    guardChecksPassed && predictedNameProb >= $(nameThreshold)
+  }
+}
+
+/**
+ * Defines static values for name identification:
+ * - Dictionary filenames and how to read them in
+ * - The number of bits to use for HyperLogLog unique entry detection (part of guard checks)
+ * - List of male/female honorifics (English only so far)
+ * - The different gender detection strategies to try
+ *
+ * Name and gender data are maintained by and taken from this repository:
+ *  https://github.com/MWYang/InternationalNames
+ * which itself sources data from:
+ *  https://ec.europa.eu/jrc/en/language-technologies/jrc-names
+ *  https://github.com/OpenGenderTracking/globalnamedata
+ *  https://github.com/first20hours/google-10000-english
+ */
+private[op] object NameDetectUtils {
+  import GenderDetectStrategy._
+
+  type NameDictionary = Set[String]
+  val DefaultNameDictionary: NameDictionary = {
+    val nameDictionary = collection.mutable.Set.empty[String]
+    val dictionaryPath = "/Names_JRC_Combined.txt"
+    val stream = getClass.getResourceAsStream(dictionaryPath)
+    val buffer = Source.fromInputStream(stream)
+    for {name <- buffer.getLines} {
+      nameDictionary += name
+    }
+    buffer.close
+    nameDictionary.toSet[String]
+  }
+  // Use the following line to use the smaller but less noisy gender dictionary as a source for names
+  // val DefaultNameDictionary: NameDictionary = DefaultGenderDictionary.value.keySet
+
+  type GenderDictionary = Map[String, Double]
+  val DefaultGenderDictionary: GenderDictionary = {
```

Comment:
same here `lazy val DefaultGenderDictionary`  - to avoid loading this every time